src/s1ap/sctp_conn.c

```diff
@@ -91,12 +92,16 @@ int recv_sctp_msg(int connSock, unsigned char *buffer, size_t len)
 			(struct sockaddr *) NULL, 0, &sndrcvinfo, &flags);
 }
 
+pthread_mutex_t sctp_sock_mutex;	/* protection to sctp_sock */
 int send_sctp_msg(int connSock, unsigned char *buffer, size_t len, uint16_t stream_no)
 {
 	uint32_t ppid = S1AP_PPID;
     uint32_t enb_fd = getEnbFdWithCbIndex(connSock);
-	return sctp_sendmsg(enb_fd, (void *)buffer, len,
+	pthread_mutex_lock(&sctp_sock_mutex);
```

Comment:
why not add lock before getting the enb-fd. the race condition might affect that also.

Reply:
I confirmed that this function is thread safe and can handle multiple threads calling this function